homeassistant/components/kodi/media_player.py

```diff
@@ -370,27 +371,42 @@ async def async_added_to_hass(self):
             return
 
         if self._connection.connected:
-            self._on_ws_connected()
-
-        self.async_on_remove(
-            async_track_time_interval(
-                self.hass,
-                self._async_connect_websocket_if_disconnected,
-                WEBSOCKET_WATCHDOG_INTERVAL,
+            await self._on_ws_connected()
+
+        def start_watchdog(event=None):
+            """Start websocket watchdog."""
+            self.async_on_remove(
+                async_track_time_interval(
+                    self.hass,
+                    self._async_connect_websocket_if_disconnected,
+                    WEBSOCKET_WATCHDOG_INTERVAL,
+                )
             )
-        )
 
-    @callback
-    def _on_ws_connected(self):
+        # If Home Assistant is already in a running state, start the watchdog
+        # immediately, else trigger it after Home Assistant has finished starting.
+        if self.hass.state == CoreState.running:
+            start_watchdog()
+        else:
+            self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, start_watchdog)
+
+    async def _on_ws_connected(self):
         """Call after ws is connected."""
         self._register_ws_callbacks()
```

Comment:
Need to make sure this is ok to call form an async context

Reply:
`_on_ws_connected()` was a `@callback` before the change. There should not have been anything in here that is not OK to call from an async context. On top of that, `_register_ws_callbacks()` is a `@callback` as well.

Reply:
good call!